feature/experimental/leader_failover_test/leader_failover_test.go

```diff
@@ -0,0 +1,371 @@
+		WithMax("203.0.113.250").
+		WithCount(250)
+
+	flow := ate.Traffic().NewFlow("Flow").
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint).
+		WithHeaders(ethHeader, ipv4Header)
+
+	ate.Traffic().Start(t, flow)
+	time.Sleep(15 * time.Second)
+	ate.Traffic().Stop(t)
+
+	time.Sleep(time.Minute)
+
+	flowPath := ate.Telemetry().Flow(flow.Name())
+
+	if !wantLoss {
+		if got := flowPath.LossPct().Get(t); got != 0 {
+			t.Errorf("FAIL: LossPct for flow named %s got %g, want 0", flow.Name(), got)
+		} else {
+			t.Logf("LossPct for flow named %s got %g, want 0", flow.Name(), got)
+		}
+	} else {
+		if got := flowPath.LossPct().Get(t); got != 100 {
+			t.Errorf("FAIL: LossPct for flow named %s got %g, want 100", flow.Name(), got)
+		} else {
+			t.Logf("LossPct for flow named %s got %g, want 100", flow.Name(), got)
+		}
+	}
+}
+
+// testArgs holds the objects needed by the test case.
+type testArgs struct {
+	ctx context.Context
+	dut *ondatra.DUTDevice
+	ate *ondatra.ATEDevice
+	top *ondatra.ATETopology
+}
+
+// testIPv4RouteAdd configures an IPV4 Entry through the given gRIBI client
+// and ensures that the entry is active by checking AFT Telemetry and Traffic.
+func testIPv4RouteAdd(ctx context.Context, t *testing.T, args *testArgs, clientA *gribi.Client) {
+
+	// Add an IPv4Entry for 203.0.113.0/24 pointing to ATE port-2 via gRIBI.
+	t.Logf("Add an IPv4Entry for %s pointing to ATE port-2 via clientA", ateDstNetCIDR)
+	clientA.AddNH(t, nhIndex, atePort2.IPv4, instance, fluent.InstalledInRIB)
+	clientA.AddNHG(t, nhgIndex, map[uint64]uint64{nhIndex: 1}, instance, fluent.InstalledInRIB)
+	clientA.AddIPv4(t, ateDstNetCIDR, nhgIndex, instance, "", fluent.InstalledInRIB)
+
+	// Verify the entry for 203.0.113.0/24 is active through AFT Telemetry.
+	t.Logf("Verify through AFT Telemetry that 203.0.113.0/24 is active")
+	ipv4Path := args.dut.Telemetry().NetworkInstance(instance).Afts().Ipv4Entry(ateDstNetCIDR)
+	if got, want := ipv4Path.Prefix().Get(t), ateDstNetCIDR; got != want {
+		t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
+	}
+
+	// Verify by running traffic that the route entry for 203.0.113.0/24 is installed and active.
+	t.Logf("Verify by running traffic that 203.0.113.0/24 is active")
+	srcEndPoint := args.top.Interfaces()[atePort1.Name]
+	dstEndPoint := args.top.Interfaces()[atePort2.Name]
+
+	testTraffic(t, args.ate, args.top, srcEndPoint, dstEndPoint, false)
+}
+
+func TestLeaderFailover(t *testing.T) {
+
+	start := time.Now()
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+
+	// Configure the DUT.
+	t.Logf("Configure DUT")
+	configureDUT(t, dut)
+
+	// Configure the ATE.
+	t.Logf("Configure ATE")
+	ate := ondatra.ATE(t, "ate")
+	top := configureATE(t, ate)
+	top.Push(t).StartProtocols(t)
+
+	t.Logf("Time check: %s", time.Since(start))
+
+	args := &testArgs{
+		ctx: ctx,
+		dut: dut,
+		ate: ate,
+		top: top,
+	}
+
+	// Set parameters for gRIBI client clientA.
+	clientA := &gribi.Client{
```

Comment:
Let's move the clientA variable into the t.Run function body for each persistence type?

Reply:
Addressed in the latest commit